Userland/Applications/Browser/main.cpp

```diff
@@ -70,14 +70,23 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     Browser::g_home_url = Config::read_string("Browser", "Preferences", "Home", "file:///res/html/misc/welcome.html");
     Browser::g_search_engine = Config::read_string("Browser", "Preferences", "SearchEngine", {});
 
-    auto ad_filter_list_or_error = Core::File::open(String::formatted("{}/BrowserContentFilters.txt", Core::StandardPaths::config_directory()), Core::OpenMode::ReadOnly);
+    auto ad_filter_list_or_error = Core::Stream::File::open(String::formatted("{}/BrowserContentFilters.txt", Core::StandardPaths::config_directory()), Core::Stream::OpenMode::Read);
     if (!ad_filter_list_or_error.is_error()) {
-        auto& ad_filter_list = *ad_filter_list_or_error.value();
-        while (!ad_filter_list.eof()) {
-            auto line = ad_filter_list.read_line();
-            if (line.is_empty())
-                continue;
-            Browser::g_content_filters.append(move(line));
+        auto maybe_ad_filter_list = Core::Stream::BufferedFile::create(ad_filter_list_or_error.release_value());
+        auto maybe_buffer = ByteBuffer::create_uninitialized(4096);
+        if (!maybe_ad_filter_list.is_error() && maybe_buffer.has_value()) {
+            auto ad_filter_list = maybe_ad_filter_list.release_value();
+            auto buffer = maybe_buffer.release_value();
+
+            auto can_read_line = ad_filter_list->can_read_line();
+            while (!can_read_line.is_error() && can_read_line.value()) {
```

Comment:
Since this is in `serenity_main`, I suggest moving this out to a new function so that you can enjoy the benefits of TRY while not exiting if anything fails. You could do something like:

```c++
if (auto result = read_content_filters(); result.is_error()) {
    dbgln("Oh no, I couldn't read the content filters: {}", result.error());
}
```

Reply:
Good idea, I did something similar for search-engines too.